contracts/chain-adapters/Ovm_CustomGasToken_L2_Forwarder.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { Arbitrum_CustomGasToken_L2_Forwarder, ICCTPTokenMessenger, FunderInterface } from "./Arbitrum_CustomGasToken_L2_Forwarder.sol";
+import { ArbitrumCustomGasTokenInbox, ArbitrumL1ERC20GatewayLike } from "../interfaces/ArbitrumBridgeInterfaces.sol";
+import { LibOptimismUpgradeable } from "@openzeppelin/contracts-upgradeable/crosschain/optimism/LibOptimismUpgradeable.sol";
+import { Lib_PredeployAddresses } from "@eth-optimism/contracts/libraries/constants/Lib_PredeployAddresses.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+/**
+ * @notice Contract containing logic to send messages from an OVM L2 to an AVM L3.
+ * @dev This contract is very similar to Arbitrum_CustomGasToken_Adapter. It is meant to bridge
+ * tokens and send messages over a bridge which uses a custom gas token, except this contract makes
+ * the assumption that it is deployed on an OpStack L2.
+ * @custom:security-contact bugs@across.to
+ */
+
+// solhint-disable-next-line contract-name-camelcase
+contract Ovm_CustomGasToken_L2_Forwarder is Arbitrum_CustomGasToken_L2_Forwarder {
```

Comment:
alternative name would be Ovm_CustomGasToken_L2_Adapter

Reply:
https://github.com/across-protocol/contracts/pull/610/commits/05dfc6eefae26f129ab68223a011766894ec144b